core/android/common/src/main/kotlin/app/k9mail/core/android/common/contact/ContactRepository.kt

```diff
@@ -10,6 +10,8 @@ interface ContactRepository {
     fun hasContactFor(emailAddress: EmailAddress): Boolean
 
     fun hasAnyContactFor(emailAddresses: List<EmailAddress>): Boolean
+
+    fun hasContactPermission(): Boolean
```

Comment:
You could use the `ContactPermissionResolver` directly, no need to also include it in the repository. This just broadens it's responsibility.

Reply:
It felt like functionality a user of `ContactRepository` would sometimes need. But I'm not set on this approach. If you think `MessageDetailsViewModel` should use `ContactPermissionResolver` directly, I'll change it.

Reply:
It is better to use it directly